### ISSUE TITLE
TreeView: wrong focus behavior with VoiceOver

### ISSUE THREAD

**Issue by silvalaura** [cengage/react-magma#1930]
**Description of bug:**
1. On MacOS, VoiceOver doesn't move to the previous parent element 
2. Wrong movement/focus behavior

Steps to reproduce:
**Bug 1**
1. Open TreeView
2. Turn on VoiceOver
3. Open the parent element with child elements
4. Move the focus to the child element
5. Move the focus back to the parent element
6. Note that the parent element is not announced (normal focus is on the parent element, VoiceOver's focus is on the child element)

**Bug 2**
1. Open TreeView
2. Open the parent element with two or more child elements
3. Open the first child element with child elements
4. Close it
5. Try to move to the next parent element using the keyboard (press the `ArrowDown` key several times)
6. Note that the focus is locked in the same parent element

[TreeView_Wrong_Behavior.webm](https://github.com/user-attachments/assets/91f87bdf-f88b-47c2-a158-b51a1d3f9008)

**Comment by nikitaorliak-cengage** [comment]
**Tested on:**
- V5: https://storybook-preview-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/treeview--no-icons
- V4: https://storybook-preview-v4-dev--upbeat-sinoussi-f675aa.netlify.app/?path=/story/treeview--no-icons

Everything works as expected. No bugs found.
